scripts/01_deploy_nft_marketplace.ts

```diff
@@ -9,18 +10,21 @@ const deployMarketplace: DeployFunction<Settings> = async (
   const nftListingTemplateResult = deployer.getDeployContractResult('NFTListing')
 
   const initialFields = {
-    nftListingTemplateId: nftListingTemplateResult.contractId,
+    nftListingTemplateId: nftListingTemplateResult.contractInstance.contractId,
     admin: network.settings.marketplaceAdminAddress,
     listingFee: network.settings.listingFee,
     commissionRate: network.settings.commissionRate
   }
 
   const result = await deployer.deployContract(NFTMarketPlace, {
     // @ts-ignore
-    initialFields: initialFields
+    initialFields: initialFields,
+    gasAmount: 100000
```

Comment:
Question: Is the `gasAmount` parameter necessary? If the specified `gasAmount` is larger than the actual gas consumption, the remaining gas will be burned.

Reply:
This is one of the cases where the estimated gas is not enough, I got `outOfGas` exception if I don't explicitly specify the gas amount.

Reply:
The reason that we have issue is because we have too many UTXOs (with tokens) for the address used for deployment. During gas estimation we take [all inputs](https://github.com/alephium/dev-alephium/blob/master/app/src/main/scala/org/alephium/app/ServerUtils.scala#L776) into consideration, which could potentially generate many outputs. Each output takes [4500](https://github.com/alephium/dev-alephium/blob/master/protocol/src/main/scala/org/alephium/protocol/vm/GasSchedule.scala#L263) gas, so the max output we can generate is 138 (`maximalGasPerTx / 4500`).

Reply:
Fixed.